src/SplitTrilinearInterpolation.jl

```diff
@@ -0,0 +1,174 @@
+module SplitTrilinearInterpolation
+using DocStringExtensions
+
+import ..Interpolation1D: Linear, Flat, interpolate1d!
+import ..BilinearInterpolation: Bilinear, interpolatebilinear!, get_dims
+
+"""
+    SplitTrilinearHV{VS, VT, HB, W1, W2}
+
+This struct stores the source and target grids and work arrays for split trilinear interpolation.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct SplitTrilinearHV{VS, VT, HB, W1, W2}
+    "Vertical source grid"
+    vsource::VS
+    "Vertical target grid"
+    vtarget::VT
+    "bilinear object containing the horizontal source and target grids"
+    bilinear::HB
+    "work array, storing fsource on the target horizontal grid"
+    fsourceintermediate::W1
+    "vertical source grid corresponding to target horizontal grid"
+    vsourceintermediate::W2
+end
+
+"""
+    SplitTrilinearHV(
+        (vsource, h1source, h2source),
+        (vtarget, h1target, h2target);
+    )
+
+A constructor for the `SplitTrilinearHV` struct.
+"""
+function SplitTrilinearHV(
+    (vsource, h1source, h2source),
+    (vtarget, h1target, h2target);
+)
+    bilinear = Bilinear(h1source, h2source, h1target, h2target)
+    nvsource = size(vsource, 1)
+    _, _, nh1target, nh2target = get_dims(bilinear)
+    fsourceintermediate = similar(vsource, nvsource, nh1target, nh2target)
+    vsourceintermediate =
+        ndims(vsource) == 1 ? vsource :
+        similar(vsource, nvsource, nh1target, nh2target)
+    ndims(vsource) == 1 ||
+        interpolatebilinear!(vsourceintermediate, bilinear, vsource)
+    return SplitTrilinearHV(
+        vsource,
+        vtarget,
+        bilinear,
+        fsourceintermediate,
+        vsourceintermediate,
+    )
+end
+
+"""
+    split_trilinear_interpolation!(
+        ftarget,
+        split_trilinear::SplitTrilinearHV,
+        fsource;
+        vextrapolate = Flat(),
+        vreverse = false,
+    )
+
+Interpolate the data `fsource` defined on a source grid onto the target grid using split trilinear interpolation.
+This specific function uses the "horizontal first" approach. The following sequence of steps are used for this 
+interpolation:
+
+1). Interpolate `fsource` onto the target horizontal grid, using blinear interpolation, at each of the vertical
+levels. Store the result in `fsourceintermediate` work array.
+
+2). If the column grid is identical for all columns, this step can be skipped and `vsourceintermediate` is identical
+to `vsource`. However, if that is not the case, use bilinear interpolation to compute the `vsourceintermediate` on
+the target horizontal grid, at each level, using bilinear interpolation.
```

Comment:
It looks like this function actually has two steps (1 and 3 here), and this step is determined during initialization/applied during what is listed as step 3 here. Maybe this explanation could go under the explanation for step 3.

I'm also not totally clear on what `vsourceintermediate` is. Is it a modification of the vertical source grid?

Reply:
Yes, and this step is handled by the constructor. I updated the documentation to clarify this!

Reply:
It looks like this point is still in the same docstring. Did you mean to remove it now that it's in the constructor?

Reply:
We have it at both places. I thought it would be good to see all steps in the process at one location!